### PR TITLE
Fixed Dlpar test.

### PR DESCRIPTION
**Pull request by smruti77**
fixed dlpar test case issue.

Signed-off-by: bismurti bidhibrata pattajoshi <bbidhibr@in.ibm.com>

### CODE REVIEW

**Comment by smruti77** [comment]
[dlpar_job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4120791/dlpar_job.log)

File: io/pci/dlpar.py [hunk 1]
```diff
@@ -88,6 +88,7 @@ def setUp(self):
         if not self.server:
             self.cancel("Managed System not got")
         self.lpar_2 = self.params.get("lpar_2", '*', default=None)
+        self.lpar2_id = self.params.get("lpar2_id", '*', default=None)
```
Comment (narasimhan-v) [inline io/pci/dlpar.py]: Why are we getting this from user, we can find this via commands.
https://github.com/avocado-framework-tests/avocado-misc-tests/blob/dbe5d1eda224d06989c9d67959eb8749ff3e5bce/io/pci/dlpar.py#L105

Comment (smruti77) [reply io/pci/dlpar.py]: ok

**Comment by smruti77** [comment]
[dlpar_job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4122282/dlpar_job.log)

File: io/pci/dlpar.py [hunk 2]
```diff
@@ -88,6 +88,12 @@ def setUp(self):
         if not self.server:
             self.cancel("Managed System not got")
         self.lpar_2 = self.params.get("lpar_2", '*', default=None)
+        cmd = 'lshwres -r io -m %s --rsubtype slot --filter lpar_names=%s ' \
+              '-F drc_index,lpar_id,drc_name,bus_id' % (self.server,
+                                                        self.lpar_2)
+        output = self.run_command(cmd)
+        for line in output:
+            self.lpar2_id = line.split(',')[1]
```
Comment (narasimhan-v) [inline io/pci/dlpar.py]: `if self.lpar_2:`

File: io/pci/dlpar.py [hunk 3]
```diff
@@ -88,6 +88,12 @@ def setUp(self):
         if not self.server:
             self.cancel("Managed System not got")
         self.lpar_2 = self.params.get("lpar_2", '*', default=None)
+        cmd = 'lshwres -r io -m %s --rsubtype slot --filter lpar_names=%s ' \
+              '-F drc_index,lpar_id,drc_name,bus_id' % (self.server,
```
Comment (narasimhan-v) [inline io/pci/dlpar.py]: displaying only lpar_id is sufficient here.

Comment (smruti77) [reply io/pci/dlpar.py]: ok

**Comment by smruti77** [comment]
JOB ID     : a7b98dc82dccea1e338b126f48984dd7fbf1ccaf
JOB LOG    : /root/avocado/job-results/job-2020-01-29T10.15-a7b98dc/job.log
 (1/3) dlpar.py:DlparPci.test_dlpar;url-7b4b: PASS (128.76 s)
 (2/3) dlpar.py:DlparPci.test_drmgr_pci;url-7b4b: PASS (85.12 s)
 (3/3) dlpar.py:DlparPci.test_drmgr_phb;url-7b4b: PASS (65.64 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 279.85 s
JOB HTML   : /root/avocado/job-results/job-2020-01-29T10.15-a7b98dc/results.html

File: io/pci/dlpar.py (hunk 2 above)
Comment (narasimhan-v) [reply io/pci/dlpar.py]: @smruti77 you need to find lpar2_id only if it is given.

Comment (smruti77) [reply io/pci/dlpar.py]: ok

Comment (smruti77) [reply io/pci/dlpar.py]: @narasimhan-v ,but self.lpar2 default is None,if self.lpar_2: this always return true.

Comment (narasimhan-v) [reply io/pci/dlpar.py]:
```
>>> a = '1'
>>> b = None
>>> bool(a)
True
>>> bool(b)
False
>>> 
```

**Comment by narasimhan-v** [comment]
@smruti77 can you upload job.log ?

**Comment by smruti77** [comment]
JOB ID     : 38f55144883656eda43c9f9314574c578103f102
JOB LOG    : /root/avocado/job-results/job-2020-01-29T07.11-38f5514/job.log
 (1/3) dlpar.py:DlparPci.test_dlpar;url-e036: PASS (112.30 s)
 (2/3) dlpar.py:DlparPci.test_drmgr_pci;url-e036: PASS (85.86 s)
 (3/3) dlpar.py:DlparPci.test_drmgr_phb;url-e036: PASS (65.77 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 264.14 s
JOB HTML   : /root/avocado/job-results/job-2020-01-29T07.11-38f5514/results.html

**Pull request merged**